### PR TITLE
University Calendar Intergration

### PR DESCRIPTION
**Pull request by jackharrhy**
- [x] get course info from subject and number
- [x] get course info for entire subject
- [x] tests for the above controllers
- [x] tests for any model methods

### CODE REVIEW

File: test/backend/models/people.spec.ts [hunk 1]
```diff
@@ -17,11 +18,17 @@ describe("backend/models/people", function () {
...
+    );
   });
 
   it("invalid nameInBanner with first word more than one character", async function () {
     const people = People.findByBannerName("Long FirstWord");
-    return expect(people).to.be.eventually.rejectedWith(BadRequest, "expected first word in nameInBanner to have a length of 1");
+    return expect(people).to.be.eventually.rejectedWith(
+      BadRequest,
+      "expected first word in nameInBanner to have a length of 1"
+    );
```
Comment (Mudkip) [inline test/backend/models/people.spec.ts:32]: Does this work? Do you not need await with eventually?

File: src/backend/scrape/calendar/listings.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  BUSI: ["BUSI-1261", "BUSI-0288", "BUSI-0401", "BUSI-0419"],
+  ED: ["EDUC-0443"],
+  HKR: ["HKR-0366"],
+  MED: ["MED-0376", "MED-0777", "MED-0784", "MED-0791", "MED-0401"],
+  MUS: ["MUSIC-0231", "MUSIC-0290", "MUSIC-0316", "MUSIC-0361"],
+  NURS: ["NURS-0230"],
+  PHAR: ["PHAR-0462"],
+  SCWK: ["SCWK-0130"],
+  MSTM: ["MARINE-0149"],
+};
```
Comment (Mudkip) [inline src/backend/scrape/calendar/listings.ts:53]: Way better. Nice.

**Review by Mudkip — APPROVED** [review]

File: test/backend/models/people.spec.ts (hunk 1 above)
Comment (jackharrhy) [reply test/backend/models/people.spec.ts:32]: if you invoke it without the await, it returns the `Promise` itself, so you hand the `Promise` to the expect, and the eventually handles awaiting the promise for you

File: src/backend/scrape/calendar/listings.ts (hunk 2 above)
Comment (jackharrhy) [reply src/backend/scrape/calendar/listings.ts:53]: stolen from @nint8835 :) https://github.com/MUNComputerScienceSociety/GarfieldBot/blob/master/plugins/CourseEmbed/listings.json

**Pull request merged**